Dev Installation: `pycryptosat` wheel, `pre-commit` requirement

Installing the developer requirements for the package fails for `pycryptosat` and `pre-commit`. In a clean virtual environment the command `python -m pip install ".[dev]"` yields the error `ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pycryptosat)`. The subsequent command `pre-commit install` returns `command not found`. Doing `pip install pre-commit` allows the command to be recognized, but`pre-commit` should of already been installed when we installed the dev requirements. 

Happy to assign myself to look closer into this, but would need some guidance on these installation tools.

## Steps to reproduce the behavior: 
1. `python -m venv venv-test`
2. `source venv-test/bin/activate`
3. `python -m pip install ".[dev]"`
4. `pre-commit install`

## Full error message for `pycryptosat`

```
Building wheel for pycryptosat (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pycryptosat (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [43 lines of output]
      running bdist_wheel
      running build
      running build_py
      running egg_info
      writing src/pycryptosat.egg-info/PKG-INFO
      writing dependency_links to src/pycryptosat.egg-info/dependency_links.txt
      writing top-level names to src/pycryptosat.egg-info/top_level.txt
      reading manifest file 'src/pycryptosat.egg-info/SOURCES.txt'
      adding license file 'LICENSE.txt'
      adding license file 'AUTHORS'
      writing manifest file 'src/pycryptosat.egg-info/SOURCES.txt'
      creating build
      creating build/lib.macosx-11.0-arm64-cpython-312
      creating build/lib.macosx-11.0-arm64-cpython-312/msvc
      copying src/msvc/stdint.h -> build/lib.macosx-11.0-arm64-cpython-312/msvc
      creating build/lib.macosx-11.0-arm64-cpython-312/picosat
      copying src/picosat/pico_config.h -> build/lib.macosx-11.0-arm64-cpython-312/picosat
      copying src/picosat/picosat.c -> build/lib.macosx-11.0-arm64-cpython-312/picosat
      copying src/picosat/picosat.h -> build/lib.macosx-11.0-arm64-cpython-312/picosat
      copying src/picosat/version.c -> build/lib.macosx-11.0-arm64-cpython-312/picosat
      creating build/lib.macosx-11.0-arm64-cpython-312/oracle
      copying src/oracle/oracle.cpp -> build/lib.macosx-11.0-arm64-cpython-312/oracle
      copying src/oracle/oracle.h -> build/lib.macosx-11.0-arm64-cpython-312/oracle
      copying src/oracle/utils.h -> build/lib.macosx-11.0-arm64-cpython-312/oracle
      running build_clib
      building 'picosatlib' library
      creating build/temp.macosx-11.0-arm64-cpython-312
      creating build/temp.macosx-11.0-arm64-cpython-312/src
      creating build/temp.macosx-11.0-arm64-cpython-312/src/picosat
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -Isrc/picosat/ -c src/picosat/picosat.c -o build/temp.macosx-11.0-arm64-cpython-312/src/picosat/picosat.o
      src/picosat/picosat.c:2282:50: warning: variable 'count_resolved' set but not used [-Wunused-but-set-variable]
        unsigned num_true, num_undef, num_false, size, count_resolved;
                                                       ^
      src/picosat/picosat.c:5334:12: warning: variable 'collect' set but not used [-Wunused-but-set-variable]
        unsigned collect, delta;
                 ^
      src/picosat/picosat.c:6467:12: warning: variable 'num_non_false' set but not used [-Wunused-but-set-variable]
        unsigned num_non_false;
                 ^
      3 warnings generated.
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64 -mmacosx-version-min=11.0 -Wno-nullability-completeness -Wno-expansion-to-defined -Wno-undef-prefix -fPIC -Werror=unguarded-availability-new -Isrc/picosat/ -c src/picosat/version.c -o build/temp.macosx-11.0-arm64-cpython-312/src/picosat/version.o
      /var/folders/m_/cksx93ys47x4621g0zbw_m4m0000gn/T/tmpsc7lpqit/tools/llvm/bin/llvm-ar rcs build/temp.macosx-11.0-arm64-cpython-312/libpicosatlib.a build/temp.macosx-11.0-arm64-cpython-312/src/picosat/picosat.o build/temp.macosx-11.0-arm64-cpython-312/src/picosat/version.o
      error: command '/var/folders/m_/cksx93ys47x4621g0zbw_m4m0000gn/T/tmpsc7lpqit/tools/llvm/bin/llvm-ar' failed: No such file or directory
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pycryptosat
Successfully built tqec
Failed to build pycryptosat
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (pycryptosat)
```

## Python dependency versions (please complete for backend bugs)

Run `pip freeze` and put the output in the `<details>` element below.

<details>
cfgv==3.4.0
distlib==0.3.8
filelock==3.15.4
identify==2.6.0
nodeenv==1.9.1
platformdirs==4.2.2
pre-commit==3.8.0
PyYAML==6.0.2
virtualenv==20.26.3
</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -16,7 +16,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
+          python-version: '3.12'
           cache: "pip"
       - name: Install dependencies
         run: |
@@ -84,7 +84,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
+          python-version: '3.12'
           cache: "pip"
       - name: Install dependencies and tqec package
         run: |

--- a/.github/workflows/exotic-oses.yml
+++ b/.github/workflows/exotic-oses.yml
@@ -1,16 +1,14 @@
 name: exotic-oses
 
 on:
-  # Trigger on PR in order to test within the PR, should be changed
-  # BEFORE merging the PR.
-  pull_request:
+  workflow_run:
+    workflows:
+      - CI
+    types:
+      - completed
     branches:
-      - main
-#   workflow_run:
-#     workflows:
-#       - CI
-#     types:
-#       - completed
+      # - main # << Not yet, test on the specific PR branch for the moment
+      - ci/test-windows-macos
 
 jobs:
   exotic-os-python-tests:

--- a/.github/workflows/exotic-oses.yml
+++ b/.github/workflows/exotic-oses.yml
@@ -1,0 +1,49 @@
+name: exotic-oses
+
+on:
+  # Trigger on PR in order to test within the PR, should be changed
+  # BEFORE merging the PR.
+  pull_request:
+    branches:
+      - main
+#   workflow_run:
+#     workflows:
+#       - CI
+#     types:
+#       - completed
+
+jobs:
+  exotic-os-python-tests:
+    strategy:
+      matrix:
+        os: ["windows-latest", "macos-latest"]
+    runs-on: ${{matrix.os}}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          # Use smallest supported Python version as the common denominator.
+          # In theory, if everything succeed in 3.10, everything should succeed in
+          # 3.11, 3.12 and follow-up version.
+          python-version: 3.10
+          cache: "pip"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.txt -r requirements-dev.txt
+      - name: Analyze code with pylint
+        run: |
+          python -m pip install pylint
+          pylint $(git ls-files '*.py')
+        continue-on-error: true
+      - name: Build
+        run: python -m pip install '.[all]'
+      - name: Test and coverage
+        run: |
+          python -m pip install pytest pytest-cov
+          python -m pytest --cov=src/tqec $(git ls-files '*_test.py')
+      - name: Mypy type checking
+        run: |
+          python -m pip install mypy
+          python -m mypy src/tqec/

--- a/.github/workflows/exotic-oses.yml
+++ b/.github/workflows/exotic-oses.yml
@@ -26,7 +26,7 @@ jobs:
           # Use smallest supported Python version as the common denominator.
           # In theory, if everything succeed in 3.10, everything should succeed in
           # 3.11, 3.12 and follow-up version.
-          python-version: 3.10
+          python-version: '3.10'
           cache: "pip"
       - name: Install dependencies
         run: |

--- a/.github/workflows/exotic-oses.yml
+++ b/.github/workflows/exotic-oses.yml
@@ -7,8 +7,7 @@ on:
     types:
       - completed
     branches:
-      # - main # << Not yet, test on the specific PR branch for the moment
-      - "ci/test-windows-macos"
+      - main
 
 jobs:
   exotic-os-python-tests:

--- a/.github/workflows/exotic-oses.yml
+++ b/.github/workflows/exotic-oses.yml
@@ -8,7 +8,7 @@ on:
       - completed
     branches:
       # - main # << Not yet, test on the specific PR branch for the moment
-      - ci/test-windows-macos
+      - "ci/test-windows-macos"
 
 jobs:
   exotic-os-python-tests:
@@ -24,7 +24,7 @@ jobs:
           # Use smallest supported Python version as the common denominator.
           # In theory, if everything succeed in 3.10, everything should succeed in
           # 3.11, 3.12 and follow-up version.
-          python-version: '3.10'
+          python-version: "3.10"
           cache: "pip"
       - name: Install dependencies
         run: |

--- a/.github/workflows/gh-pages.yml
+++ b/.github/workflows/gh-pages.yml
@@ -18,7 +18,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.12
+          python-version: '3.12'
           cache: "pip"
       - name: Install dependencies
         run: |

--- a/src/tqec/sketchup/collada_test.py
+++ b/src/tqec/sketchup/collada_test.py
@@ -1,3 +1,4 @@
+import os
 import tempfile
 
 import pytest
@@ -38,10 +39,15 @@ def test_logical_cnot_collada_write_read(pipe_length: float) -> None:
 
     block_graph = zx_graph.to_block_graph()
 
-    with tempfile.NamedTemporaryFile(suffix=".dae") as temp_file:
+    # Set `delete=False` to be compatible with Windows
+    # https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile
+    with tempfile.NamedTemporaryFile(suffix=".dae", delete=False) as temp_file:
         block_graph.to_dae_file(temp_file.name, pipe_length)
         block_graph_from_file = BlockGraph.from_dae_file(
             temp_file.name, "Logical CNOT Block Graph"
         )
         assert block_graph_from_file == block_graph
         assert block_graph_from_file.to_zx_graph() == zx_graph
+
+    # Manually delete the temporary file
+    os.remove(temp_file.name)

